### PR TITLE
Graduate JobTrackingWithFinalizers to beta in 1.23

### PR DESCRIPTION
**Pull request by alculquicondor**
Ref #2307

Updates graduation criteria and completes PRR.

### CODE REVIEW

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (alculquicondor) [inline keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:314]: @lavalamp this is what we discussed offline. This is the behavior I would like, but maybe this is the wrong approach due to backwards compatibility. Let me know if you have better suggestions.

Comment (lavalamp) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:314]: 5000/60 = 83 QPS; if each request is 100ms, that's a concurrency of ~9, which is not very much. You probably don't even need to adjust the default APF levels (you do need to adjust or turn off the client side rate limit).

Note that pod creations have a bunch of follow-on requests and there's not much point in giving this special priority to create them if the rest of the system can't then keep up.

What you should do instead is look at the shares given to workload-high and the other PLs and figure out what overall in-flight limit needs to be such that workload high ends up with at least that much concurrency, and document that, so operators know how big of a cluster they need to get that throughput.

Note that to get your maximum throughput, you need to multiplex your requests across all apiservers, which is challenging in some deployments. But it may be more important than e.g. making a new PL.

Comment (alculquicondor) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:314]: If I determine that we need an update of the existing PLs, should that be (and can it be) dependent on the feature gate for this KEP?

Comment (lavalamp) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:314]: That would be very awkward, so I'd recommend no. Not every k8s cluster needs this to go this fast, so I think in that case we'd most likely ask you to just document what needs to be done. But let's see what that is before making any decisions.

Comment (wojtek-t) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:337]: > I think we are good with the current default APF configuration. And we can recommend users to set the client QPS to 100 if they expect to have 2500 pods finishing per minute or disable it completely if they expect to process more than 5k pods per minute.

SGTM

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (wojtek-t) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:338]: +1 for removing it - please proceed

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (wojtek-t) [inline keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:420]: nit: move to preview line

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (wojtek-t) [inline keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:456]: Can you please ack that those will get run before feature will graduate? And that the KEP will get updated with their results [sth like: https://github.com/kubernetes/enhancements/pull/2538 ]

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (wojtek-t) [inline keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:466]: I think some metric would be useful here. How about a counter summarizing how many pod additions to uncountedTerminatedPods were done?

**Review by wojtek-t — COMMENTED** [review]
I'm fine with it modulo the comments I added.

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (alculquicondor) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md]: Ack

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (alculquicondor) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:465]: Added, but rather when they are removed from `uncountedTerminatedPods`

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (alculquicondor) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md]: Done

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (alculquicondor) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md]: Updated.

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (wojtek-t) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:465]: Sure - that's better.

**Comment by wojtek-t** [comment]
@alculquicondor - I'm generally fine with this; please work with SIG approvers to have that approved (@soltysh :) ) and I will approve after this

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (wojtek-t) [inline keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:431]: Have those been added? If not, can we add a similar comment as for upgrade tests?

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (wojtek-t) [inline keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:452]: How about enablement?

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (wojtek-t) [inline keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:454]: prior to Beta graduation

**Review by wojtek-t — COMMENTED** [review]
Just some minor comments - LGTM after applied.

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (alculquicondor) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:431]: Yes, we have them. Updated.

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (alculquicondor) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md]: That too. Updated.

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (alculquicondor) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md]: Fixed

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (wojtek-t) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:432]: Can you please add links?

**Review by wojtek-t — COMMENTED** [review]
Sorry - one last nit

File: keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md
Comment (alculquicondor) [reply keps/sig-apps/2307-job-tracking-without-lingering-pods/README.md:432]: Done :)

**Pull request merged**
